cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -509,3 +511,96 @@ func (s *Server) CreateInstalledPackage(ctx context.Context, request *corev1.Cre
+	log.Infof("+kapp-controller UpdateInstalledPackage")
+
+	// Validate the request
+	if request == nil || request.GetInstalledPackageRef() == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "no request AvailablePackageRef provided")
+	}
+	if request.GetInstalledPackageRef().GetContext().GetNamespace() == "" || request.GetInstalledPackageRef().GetIdentifier() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "required context or identifier not provided")
+	}
+	if request.GetInstalledPackageRef().GetIdentifier() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "no request Name provided")
+	}
+
+	packageRef := request.GetInstalledPackageRef()
+	cluster := packageRef.GetContext().GetCluster()
+	reconciliationOptions := request.GetReconciliationOptions()
+	pkgVersion := request.GetPkgVersionReference().GetVersion()
+	installedPackageName := request.GetInstalledPackageRef().GetIdentifier()
+	values := request.GetValues()
+	targetNamespace := request.GetInstalledPackageRef().GetContext().GetNamespace()
+
+	if cluster == "" {
+		cluster = s.globalPackagingCluster
+	}
+
+	typedClient, _, err := s.GetClients(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	// fetch the package install
+	pkgInstall, err := s.getPkgInstall(ctx, cluster, targetNamespace, installedPackageName)
+	if err != nil {
+		return nil, errorByStatus("get", "PackageInstall", installedPackageName, err)
+	}
+
+	// Update the values.yaml values file if any is passed, otherwise, delete the values
+	if values != "" {
+		secret, err := s.newSecret(installedPackageName, values, targetNamespace)
+		if err != nil {
+			return nil, errorByStatus("upsate", "Secret", secret.Name, err)
+		}
+		updatedSecret, err := typedClient.CoreV1().Secrets(targetNamespace).Update(ctx, secret, metav1.UpdateOptions{})
+		if updatedSecret == nil || err != nil {
+			return nil, errorByStatus("update", "Secret", secret.Name, err)
+		}
+	} else {
+		// Delete all the associated secrets
+		for _, packageInstallValue := range pkgInstall.Spec.Values {
+			secretId := packageInstallValue.SecretRef.Name
+			err := typedClient.CoreV1().Secrets(targetNamespace).Delete(ctx, secretId, metav1.DeleteOptions{})
+			if err != nil {
+				return nil, errorByStatus("delete", "Secret", secretId, err)
+			}
+		}
+	}
+	// Update the rest of the fields
+	pkgInstall.Spec.PackageRef.VersionSelection = &vendirVersions.VersionSelectionSemver{Constraints: pkgVersion}
+	if reconciliationOptions != nil {
+		if reconciliationOptions.Interval > 0 {
+			pkgInstall.Spec.SyncPeriod = &metav1.Duration{Duration: time.Duration(reconciliationOptions.Interval) * time.Second}
+		}
+		if reconciliationOptions.ServiceAccountName != "" {
+			pkgInstall.Spec.ServiceAccountName = reconciliationOptions.ServiceAccountName
+		}
+		pkgInstall.Spec.Paused = reconciliationOptions.Suspend
+	}
+
+	// get rid of the status field, since now there will be a new reconciliation process
```

Comment:
Is that necessary? I had thought the status section of a resource was only updated by the controller... At least, if I edit the status of a resource via kubectl, it appears to just be ignored (or is very quickly set back, not sure).

Reply:
I'm not sure, I must reckon... I just assumed I should do it as per the snippet from Flux's code (below) to force a reconciliation trigger, but I guess that logic does not make sense here, as kapp-ctrl should be handling every state change and Kubeapps shouldn't be messing around with it.
Deleting it right away!

https://github.com/kubeapps/kubeapps/blob/260fb34fd70cc5d00faf158b517820c8cc34e845/cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go#L458

Reply:
Yeah - as above, I'm not sure. I just know what happens via the kubectl cli if I try to modify the status. YMMV